compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -542,19 +606,57 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         closure_span: Span,
         closure_clause: hir::CaptureBy,
         min_captures: Option<&ty::RootVariableMinCaptureList<'tcx>>,
-    ) -> Vec<hir::HirId> {
+        should_do_drop_migration: bool,
```

Comment:
this would need changing the doc string

Reply:
I would keep the original funciton as is and create a new one. They don't share too much code together, and it just requires maintaining cotext of the conditionals while reading the code

Reply:
I mean, I feel like it would need to be renamed in that case

Reply:
Can we split this function into two? Compute 2229_drop_migrations and 2229_auto_trait_migrations